update-vo-config/update-vo-config

```diff
@@ -0,0 +1,507 @@
+        template.write("        'description', '%s',\n" % mapping['desc'])
+        template.write("        'fqan', '%s',\n" % mapping['expr'])
+        template.write("        'suffix', '%s',\n" % suffix)
+        template.write("        'suffix2', '%s',\n" % suffix2)
+        template.write("    ),\n")
+
+    template.write(");\n\n")
+
+    # Base UID
+    base_uid = vo_details['id'] * 1000
+    template.write("'base_uid' ?= %i;\n" % base_uid)
+    template.close()
+
+
+def update_voms_servers(vo_details, basedir, namespaces, debug=False):
+    """write template containing the voms certficate
+    """
+    voms_servers = {}
+    for vo_config in vo_details:
+        if not vo_config['voms_endpoints']:
+            sys.stderr.write("No VOMS server defined in any VO\n")
+        else:
+            for server in vo_config['voms_endpoints']:
+                if server.hostname not in voms_servers:
+                    voms_servers[server.hostname] = server.cert
+    for (hostname, cert) in voms_servers.items():
+        server_cert_ns = namespaces['cert_dir'] + os.path.sep + hostname
+        server_cert_tpl = basedir + os.path.sep + server_cert_ns + ".pan"
+        if cert is None:
+            sys.stdout.write("No certicate defined for %s " % hostname +
+                             "voms server.\n")
+            x509_cert = None
+        else:
+            try:
+                x509_cert = crypto.load_certificate(crypto.FILETYPE_PEM, cert)
+
+            except crypto.Error as e:
+                sys.stderr.write("ERROR Certificate for %s " % hostname +
+                                 "could not be loaded: %s\n" % str(e))
+        old_x509_cert = None
+        if x509_cert is not None and os.path.isfile(server_cert_tpl):
+            if debug:
+                sys.stdout.write("DEBUG Retrieving VOMS server"
+                                 " %s existing certificate\n" % hostname)
+            template = open(server_cert_tpl, 'r')
+            old_cert = ""
+            in_cert_part = False
+            cert_start_reg = re.compile("^'cert' \?=")
+            cert_end_reg = re.compile("^EOF")
+            for line in template.readlines():
+                if in_cert_part:
+                    if re.search(cert_end_reg, line):
+                        in_cert_part = False
+                        break
+                    else:
+                        old_cert += line
+                elif re.search(cert_start_reg, line):
+                    in_cert_part = True
+            template.close()
+            if old_cert:
+                try:
+                    old_x509_cert = crypto.load_certificate(
+                        crypto.FILETYPE_PEM, old_cert
+                    )
+                except crypto.Error as e:
+                    sys.stderr.write("ERROR Existing certificate for " +
+                                     "%s could not be " % hostname +
+                                     "loaded: %s\n" % str(e))
```

Comment:
Ditto.

Reply:
The str call has been removed in commit 091e53d42df1e799fe46897c9ff4c91567124ada.